codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/MainSourceGenerator.scala

```diff
@@ -0,0 +1,248 @@
+ */
+object MainSourceGenerator {
+
+  import com.lightbend.akkasls.codegen.SourceGeneratorUtils._
+
+  def generateUnmanaged(model: ModelBuilder.Model): Iterable[File] = {
+    val generatedSources = Seq.newBuilder[File]
+
+    val mainPackage = mainPackageName(model.services.keys ++ model.entities.keys)
+    val mainPackageStr = mainPackage.mkString(".")
+    val mainPackagePath = mainPackage.mkString("/")
+    val mainClassName = "Main"
+
+    generatedSources += File(
+      s"$mainPackagePath/$mainClassName.scala",
+      mainSource(mainPackageStr, mainClassName, model.entities, model.services))
+
+    generatedSources.result()
+  }
+
+  def generateManaged(model: ModelBuilder.Model): Iterable[File] = {
+    val generatedSources = Seq.newBuilder[File]
+
+    val mainPackage = mainPackageName(model.services.keys ++ model.entities.keys)
+    val mainPackageStr = mainPackage.mkString(".")
+    val mainPackagePath = mainPackage.mkString("/")
+
+    generatedSources += File(
+      s"$mainPackagePath/AkkaServerlessFactory.scala",
+      akkaServerlessFactorySource(mainPackageStr, model))
+
+    generatedSources.result()
+  }
+
+  private[codegen] def mainSource(
+      mainClassPackageName: String,
+      mainClassName: String,
+      entities: Map[String, Entity],
+      services: Map[String, Service]): String = {
+
+    val entityImports = entities.values.collect {
+      case entity: ModelBuilder.EventSourcedEntity => entity.fqn.fullQualifiedName
+      case entity: ModelBuilder.ValueEntity        => entity.fqn.fullQualifiedName
+      case entity: ModelBuilder.ReplicatedEntity   => entity.fqn.fullQualifiedName
+    }.toSeq
+
+    val serviceImports = services.values.collect {
+      case service: ModelBuilder.ActionService => service.classNameQualified
+      case view: ModelBuilder.ViewService      => view.classNameQualified
+    }.toSeq
+
+    val componentImports =
+      generateImports(Iterable.empty, mainClassPackageName, entityImports ++ serviceImports, semi = false)
+    val entityRegistrationParameters = entities.values.toList
+      .sortBy(_.fqn.name)
+      .collect {
+        case entity: ModelBuilder.EventSourcedEntity => s"new ${entity.fqn.name}(_)"
+        case entity: ModelBuilder.ValueEntity        => s"new ${entity.fqn.name}(_)"
+        case entity: ModelBuilder.ReplicatedEntity   => s"new ${entity.fqn.name}(_)"
+      }
+
+    val serviceRegistrationParameters = services.values.toList
+      .sortBy(_.fqn.name)
+      .collect {
+        case service: ModelBuilder.ActionService => s"new ${service.className}(_)"
+        case view: ModelBuilder.ViewService      => s"new ${view.className}(_)"
+      }
+
+    val registrationParameters = entityRegistrationParameters ::: serviceRegistrationParameters
+    s"""|package $mainClassPackageName
+        |
+        |import com.akkaserverless.scalasdk.AkkaServerless
+        |import org.slf4j.LoggerFactory
+        |${componentImports}
```

Comment:
perhaps add those two extra imports in `generateImports`?

Reply:
thanks, done in 94cccf7a5e725516a27f4eb7e150e6949bb8a9f9